packages/ai-ide/src/common/coder-replace-prompt-template.ts

```diff
@@ -51,7 +51,11 @@ Instead, for each file you want to propose changes for:
 ${withSearchAndReplace ? ' If ~{changeSet_replaceContentInFile} continously fails use ~{changeSet_writeChangeToFile}. Calling a function on a file will override previous \
 function calls on the same file, so you need exactly one successful call with all proposed changes per changed file. The changes will be presented as a applicable diff to \
 the user in any case.' : ''}
-  
+
+## Previous Interactions and Background
```

Comment:
1. I think this is too specific, because we can also use these new variables for new tasks
2. We should not put this heading here, but inside of the variable resolvement. If there is not context from this variable, the heading should also not appear.

Reply:
Tested it a bit, we should put this at the end of the prompt (plus the comments above)